components/Tabs2/src/MDCTabBarView.m

```diff
@@ -16,4 +16,51 @@
 
 @implementation MDCTabBarView
 
+#pragma mark - Properties
+
+- (void)setItems:(NSArray<UITabBarItem *> *)items {
+  NSAssert([NSThread isMainThread], @"Item array may only be set on the main thread");
```

Comment:
Material iOS UI components don't explicitly guard properties from being accessed off the main thread. This is because UIKit is generally not thread safe to start with. Is there a strong technical reason for this assertion? If not, please remove it.

Reply:
+1 to remove - this was copied from MDCItemBar.  It's a good instinct to copy from existing code that seems to do the same thing, but unfortunately that's old code (pre-MDC) and doesn't really follow our team's current conventions.

Reply:
Sure. I will try to understand more on how current stand instead of copying from old existing code. :)